l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -221,6 +523,60 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
         return balanceAfter - balanceBefore;
     }
 
+    function bridgehubDeposit(
+        uint256 _chainId,
+        address _prevMsgSender,
+        bytes calldata _data
+    ) external payable override onlyBridgehub returns (L2TransactionRequestTwoBridgesInner memory request) {
+        (address _l1Token, uint256 _amount, address _l2Receiver) = abi.decode(_data, (address, uint256, address));
+        require(msg.value == 0, "L1EB m.v > 0 for BH dep");
+        require(l2BridgeAddress[_chainId] != address(0), "L1EB b. n deployed");
+        require(bridgehub.baseToken(_chainId) != _l1Token, "L1EB base deposit");
+
+        require(_amount != 0, "6T"); // empty deposit amount
+        uint256 amount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _amount);
+        require(amount == _amount, "5T"); // The token has non-standard transfer logic
+
+        if (!hyperbridgingEnabled[_chainId]) {
+            chainBalance[_chainId][_l1Token] += _amount;
+        }
+        bytes32 txDataHash = keccak256(abi.encode(_prevMsgSender, _l1Token, _amount));
+
+        {
+            // Request the finalization of the deposit on the L2 side
+            bytes memory l2TxCalldata = _getDepositL2Calldata(msg.sender, _l2Receiver, _l1Token, _amount);
+
+            request = L2TransactionRequestTwoBridgesInner({
+                l2Contract: l2BridgeAddress[_chainId],
+                l2Calldata: l2TxCalldata,
+                factoryDeps: new bytes[](0),
+                txDataHash: txDataHash
+            });
+        }
+        emit BridgehubDepositInitiatedSharedBridge(
+            _chainId,
+            txDataHash,
+            _prevMsgSender,
+            _l2Receiver,
+            _l1Token,
+            _amount
+        );
+        if (_chainId == ERA_CHAIN_ID) {
+            // kl todo. Should emit this event here? Should we not allow this method for era?
+            emit DepositInitiated(0, _prevMsgSender, _l2Receiver, _l1Token, _amount);
```

Comment:
@vladbochok @StanislavBreadless  Note that the event has to change here. Because of the new bridging process and requestL2TransactionTwoBridges, we cannot emit the required event.

If we want to be backwards comptible we have to stop Era from using this method. I don't like that solution. Wdyt?

Reply:
Do you mean that tx hash is zero here?

Reply:
Closed, as Vlad said we don't need this level of backward compatibility